client/src/pages/queries/components/chart-detail.tsx

```diff
@@ -0,0 +1,284 @@
+import React, {
+  useMemo,
+  cloneElement,
+  useContext,
+  useState,
+  useCallback,
+} from 'react'
+import { Result, Form, Select, Input, Button, Modal, message } from 'antd'
+import { range } from 'lodash-es'
+import { useMutation, useQuery, useQueryClient } from 'react-query'
+import EditorContext from 'contexts/editor-context'
+import { Chart, ChartTypes } from 'types/chart'
+import {
+  deleteChart,
+  fetchChart,
+  fetchChartsForQuery,
+  updateChart,
+} from '../chart-queries'
+import { chartGroups, chartList, chartOrder } from '../data/chart-list'
+
+type ChartDetailProps = {
+  chartId: number
+  changeActiveChartId: (id: number) => void
+}
+
+export default function ChartDetail({
+  chartId,
+  changeActiveChartId,
+}: ChartDetailProps) {
+  const { queryResult, queryId } = useContext(EditorContext)
+  const [form] = Form.useForm()
+
+  const [chartType, setChartType] = useState<ChartTypes>('line')
+  const [label, setLabel] = useState<string | undefined>()
+  const [values, setValues] = useState<string[] | undefined>()
+  const [title, setTitle] = useState<string>('Untitled Chart')
+
+  const { data: chart, isLoading, error } = useQuery(
+    ['chart', chartId],
+    () => fetchChart(chartId),
+    {
+      onSettled: (chart) => {
+        if (chart) {
+          setTitle(chart.name)
+          setValues(chart.values)
+          setLabel(chart.label)
+          setChartType(chart.type)
+          form.setFieldsValue({
+            type: chart.type,
+            name: chart.name,
+            label: chart.label,
+            values: chart.values,
+          })
+        }
+      },
+    },
+  )
+  const [disabled, setDisbaled] = useState<boolean>(true)
+
+  const { data: charts } = useQuery(['charts', queryId], () =>
+    fetchChartsForQuery(parseInt(queryId)),
+  )
+
+  const queryClient = useQueryClient()
+
+  const { mutate: editChart, isLoading: isUpdatingChart } = useMutation(
+    updateChart,
+    {
+      onMutate: (updatedChart) => {
```

Comment:
Instead of using `onMutate` use `onSuccess`. If we are using `onMutate`, then we should also revert back to the old data on `onError`. https://react-query.tanstack.com/guides/optimistic-updates#updating-a-list-of-todos-when-adding-a-new-todo